RPS.js

```diff
@@ -15,127 +15,86 @@ function getCpuChoice () {
     return cpuChoice;
 }
 
-const rockBtn = document.getElementById('rock');
-rockBtn.addEventListener('click', rockChoice);
-function rockChoice () {
-    console.log('The function rockChoice() is running....');
-    cpuButtonHighlightRemoval(cpuRock);
-    cpuButtonHighlightRemoval(cpuPaper);
-    cpuButtonHighlightRemoval(cpuScissors);
-    let winnerOfRound = '';
-    let cpuChoice = getCpuChoice();
-    let playerChoice = 'rock';
-    if (playerChoice === cpuChoice) {
-        winnerOfRound = 'Draw, there is no winner';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuRock);
-        playerCPUDraw(playerScore, cpuScore); 
-        return changeBannerText(winnerOfRound);
-    }
-    else if (playerChoice == 'rock' && cpuChoice === 'paper') {
-        winnerOfRound = 'You lose! Paper beats rock!';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuPaper);
-        cpuIncrementScore(cpuScore, cpuScoreElement);
-        return changeBannerText(winnerOfRound);
-    }
-    else if (playerChoice == 'rock' && cpuChoice === 'scissors') {
-        winnerOfRound = 'You win! Rock beats scissors!';
-        console.log(winnerOfRound);
-        cpuButtonHighlight(cpuScissors);
-        playerIncrementScore(playerScore, playerScoreElement);
-        return changeBannerText(winnerOfRound);
+
+// 0 == tie, 1 == victory, -1 == loss
+const resultMatrix = [
+                                // CPU
+                        // rock, paper, scissors
+    /*Player   rock     */[ 0,     -1,      1],
+    /*         paper    */[ 1,      0,     -1],
+    /*         scissors */[-1,      1,      0]
+
+]
+
+function determineResult(player, cpu) {
+    // rock = 0, paper = 1, scissors = 2
+    const choice = {
+        "rock": 0,
+        "paper": 1,
+        "scissors": 2
     }
```

Comment:
this choice to integer mapping I'm using isn't so great. There's a type of data structure called an ENUM in other languages including TypeScript that would have been a good fit of mapping. They're generally a good strategy when you want to represent something using an integer, which is generally more robust than using strings, which can be prone to error.

Reply:
https://stackoverflow.com/questions/4709175/what-are-enums-and-why-are-they-useful